### ISSUE TITLE
Cursor flickering when entering text inside of a tmux session

### ISSUE THREAD

**Issue by antmelon** [jesseduffield/lazygit#3434]
**Describe the bug**
When using a text form such as the commit popup inside of a tmux session, the cursor will flicker at random points on the gui while typing. Outside of a tmux session this behavior is not seen

**To Reproduce**

1. Launch tmux session
2. Open Lazygit and enter a text popup such as the commit form
3. When typing screen tare/cursor flicker can be seen

**Expected behavior**
No random cursor flashing should be happening on GUI

**Version info:**
 _lazygit commit=3675570a391b1a49ddd198b4c7e71e17701d4404, build date=2024-03-23T09:09:11Z, build source=binaryRelease, version=0.41.0, os=linux, arch=amd64, git version=2.25.1_
_git version 2.25.1_

**Comment by laurentS** [comment]
I'm observing the same issue, both with lazygit 0.41 and 0.42, both downloaded from the release page (`linux_x84_64` builds), running on debian sid. I did not see this problem before, so it's likely a regression in 0.41. As OP, I only see the flickering inside tmux. I use alacritty as a term, but that's probably unrelated.

Current lazygit version: `commit=6fcb7eb8bb616c170506312870b3bf15f3dbe37c, build date=2024-05-19T10:15:28Z, build source=binaryRelease, version=0.42.0, os=linux, arch=amd64, git version=2.43.0`

**Comment by Kaarel** [comment]
Same experience on Windows Terminal. Typing into the "Commit summary" is laggy and flickery. See the attached gif. It's not bad recording quality or slow network, this is how it actually looks and feels.

Lazygit version: `commit=6fcb7eb8bb616c170506312870b3bf15f3dbe37c, build date=2024-05-19T10:15:28Z, build source=binaryRelease, version=0.42.0, os=windows, arch=amd64, git version=2.45.2.windows.1`

![Lazygit laggy typing](https://github.com/jesseduffield/lazygit/assets/108327/c406006c-15d1-497b-a83d-8561a0c9eb0c)

**Comment by peterlandry** [comment]
Same issue with zellij on Mac.

**Comment by juanibiapina** [comment]
It seems to me @antmelon and @Kaarel are reporting two different symptoms. I have the flickering reported by @antmelon when running inside `ghostty -> tmux` (but not `alacritty -> tmux`), but the typing slowdown reported by @Kaarel happens when running inside `alacritty -> tmux -> nvim terminal` in NixOS only (since I don't care about this case I didn't do further tests). The two issues may not be related.

**Comment by stefanhaller** [comment]
Could you test this again with current master? It may (or may not) be improved by the merge of #4166.

**Comment by Kaarel** [comment]
No change in behaviour for me.

lazygit --version:
commit=91cb1ff29a01fdf643ccc38b97fd816b2a097bec, build date=2025-01-11T04:06:07Z, build source=binaryRelease, version=0.45.0, os=windows, arch=amd64, git version=2.47.1.windows.1

Windows Terminal:
Version: 1.21.3231.0

Note that the typing issue (illustrated by the image above) is present in LazyGit when I run it from Windows Terminal, cmd, Mingw64 based Git Bash (comes with Git for Windows). Typing in each of these respective shell prompts is fine. Typing in other applications (eg neovim, yazi) run from within each of these shells also has no issues. Only seems to affect LazyGit.

**Comment by juanibiapina** [comment]
Both bugs still happen on master.

Btw, it ran so smoothly! This project is a great inspiration for me.

**Comment by GitMurf** [comment]
@Kaarel  same issues for me in windows terminal! I thought it was just “my computer” so I’ve just put up with it. But recently I tried lazygit in wsl2 via windows terminal and it is buttery smooth and both issues mentioned above are non existent.

Can you try wsl via windows terminal and see if it also works fine for you?

**Comment by GitMurf** [comment]
> In WSL it works fine. Native Windows version has the issue. I have git repos in Windows and in WSL.

Yep same exact thing for me. And it only started the last couple months.

**Comment by spencer-scw** [comment]
> Same issue with zellij on Mac.

I'm noticing this too, but only when a fetch is ongoing. I suspect that the draw rate on the fetch progress spinner is interfering with the cursor's position and causing a flickery experience. This happens for any text entry field (commit messages, filter searches, etc.) I'm using the latest build of Lazygit in Zellij / Alacritty / MacOS.

**Comment by sebastianstudniczek** [comment]
Check whether enabling `synchronized input` for tmux does solve your issue. I stumbled across this issue when searching for such cursor flickering in neovim.

```bash
tmux set-option -g terminal-features ",xterm-256color:Sync,tmux-256color:Sync"
```

https://github.com/wezterm/wezterm/issues/7629
